visualization/app/codeCharta/ui/heightSettingsPanel/heightSettingsPanel.component.ts

```diff
@@ -70,6 +94,14 @@ export class HeightSettingsPanelController
 		this.applyDebouncedTopLabels()
 	}
 
+	applySettingsMetricLabelValueChanged(){
```

Comment:
I have a question. We subscribe the onShowMetricLabelNameValueChanged and onShowMetricLabelNodeNameChanged. Now, we are calling storeService.dispatch(....

Wouldn't the ShowMetricLabelNameValueService and ShowMetricLabelNodeNameService be called from the resulting onStoreChanged event and then they would broadcast the appropriate events that the label settings have changed? 

Just for my understanding: A roundtrip would be the result:
HeightSettingsPannel dispatch new state > State dispatch onStoreChanged > LabelServices notify subscriber > HeightSettingsPannel is called (onNameValue and onNameName changed)

This comment is not really concerning your PR rather than the event architecture.

Reply:
You are correct, and this change is then recognized as a change of state by angular amd the component gets "re-rendered".
You can alos refer to the graphic under "State Managment" under https://maibornwolff.github.io/codecharta/docs/new-to-code/ for a nice visualization